### PR TITLE
 Replace bash function set_faillock_option by a Jinja macro

### PR DESCRIPTION
**Pull request by jan-cerny**
#### Description:

Replaces bash function set_faillock_option by Jinja macro bash_set_faillock_option. Include the common code for all 3 remediations to the macro.

#### Rationale:

This is another part of our effort to convert Bash functions to Jinja macros. Once all macros will be converted we will be able to remove the legacy code and confusing sourcing of non-existent file.

### CODE REVIEW

**Comment by matejak** [comment]
Please rebase to resolve conflicts.

**Comment by jan-cerny** [comment]
@matejak I have rebased and resolved conflicts.

**Comment by matejak** [comment]
We had nice functions before, do you think that you could define some "private" macros, so at least the remediation source indicates that there are three things going on?  Now it is just a big bunch of code and although there are comments, it is good to know that the remediation consists of three independent parts.

File: shared/macros-bash.jinja [hunk 1]
```diff
@@ -319,6 +319,57 @@ else
...
 {{%- endmacro -%}}
 
+{{#
+# Sets PAM faillock module options and values.
+# Also it adds pam_faillock.so as required module for account.
+# option: faillock option eg. deny, unlock_time
+# value: value of option
+#}}
+{{%- macro bash_set_faillock_option(option, value) -%}}
+AUTH_FILES[0]="/etc/pam.d/system-auth"
```
Comment (matejak) [inline shared/macros-bash.jinja]: I would rather use `AUTH_FILES=("/etc/pam.d/system-auth" "/etc/pam.d/password-auth")`

File: shared/macros-bash.jinja [hunk 2]
```diff
@@ -319,6 +319,57 @@ else
...
+AUTH_FILES[1]="/etc/pam.d/password-auth"
+
+for pam_file in "${AUTH_FILES[@]}"
+do
+  # is auth required pam_faillock.so preauth present?
+  if grep -qE "^\s*auth\s+required\s+pam_faillock\.so\s+preauth.*$" "$pam_file" ; then
+    # is the option set?
+    if grep -qE "^\s*auth\s+required\s+pam_faillock\.so\s+preauth.*{{{ option }}}=([0-9]*).*$" "$pam_file" ; then
+      # just change the value of option to a correct value
+      sed -i --follow-symlinks "s/\(^auth.*required.*pam_faillock.so.*preauth.*silent.*\)\({{{ option }}} *= *\).*/\1\2{{{ value }}}/" "$pam_file"
```
Comment (matejak) [inline shared/macros-bash.jinja]: Please use single quotes whenever you can, as inside double quotes, the backslash should be escaped twice, which may be confusing.

**Comment by jan-cerny** [comment]
@matejak I have improved it by splitting out 3 low level macros. I have used  diffrent array definition and I also replaced double quotes by single quotes.

```
[jcerny@thinkpad tests{bash_set_faillock_option}]$ python3 test_suite.py rule --libvirt qemu:///system ssgts_rhel8 --datastream ../build/ssg-rhel8-ds.xml accounts_passwords_pam_faillock_deny
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/jcerny/scap-security-guide/tests/logs/rule-custom-2019-11-21-1642/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_accounts_passwords_pam_faillock_deny
ERROR - Script skip_longer_comment.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp found issue:
ERROR - Rule evaluation resulted in pass, instead of expected fail during initial stage 
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_accounts_passwords_pam_faillock_deny'.
INFO - Script config_without_skip.pass.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
INFO - Script remediable_sssd_authconfig.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
INFO - Script skip_correctly.pass.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
INFO - Script default_die_pam_unix.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
INFO - Script skip_longer.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
INFO - Script disablefaillock_authconfig.fail.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
INFO - Script skip_correctly_short.pass.sh using profile xccdf_org.ssgproject.content_profile_ospp OK
```

Unfortunately, the fail of scenario `skip_longer_comment.fail.sh` isn't caused by this PR because of https://bugzilla.redhat.com/show_bug.cgi?id=1549671

**Comment by matejak** [comment]
Great improvement, thanks!

**Pull request merged**